php-frontend/src/test/java/org/sonar/php/parser/declaration/ClassVariableDeclarationTest.java

```diff
@@ -19,19 +19,27 @@
  */
 package org.sonar.php.parser.declaration;
 
-import static org.sonar.php.utils.Assertions.assertThat;
-
 import org.junit.Test;
 import org.sonar.php.parser.PHPLexicalGrammar;
 
+import static org.sonar.php.utils.Assertions.assertThat;
+
 public class ClassVariableDeclarationTest {
 
   @Test
   public void test() {
-    assertThat(PHPLexicalGrammar.CLASS_VARIABLE_DECLARATION)
+     assertThat(PHPLexicalGrammar.CLASS_VARIABLE_DECLARATION)
       .matches("var $a;")
       .matches("var $a, $b;")
       .matches("public static $a;")
+      .matches("public int $id;")
+      .matches("protected ClassName $classType;")
+      .matches("public static iterable $staticProp;")
+      .matches("var bool $flag;")
+      .matches("public string $str = \"foo\";")
+      .matches("public float $x, $y;")
+      .matches("public self $x;")
+      .matches("public parent $x;")
       .notMatches("public static var $a;")
```

Comment:
I suggest that we add tests with `object` and `array` to be exhaustive about built-in types :blush:.